scripts/test/Muon/grouping_tab/pairing_table_group_selector_test.py

```diff
@@ -0,0 +1,164 @@
+import unittest
+import sys
+
+if sys.version_info.major == 3:
+    from unittest import mock
+else:
+    import mock
+
+from PyQt4 import QtGui
+
+from Muon.GUI.Common.pairing_table_widget.pairing_table_widget_model import PairingTableModel
+from Muon.GUI.Common.pairing_table_widget.pairing_table_widget_view import PairingTableView
+from Muon.GUI.Common.pairing_table_widget.pairing_table_widget_presenter import PairingTablePresenter
+
+from Muon.GUI.Common.muon_group import MuonGroup
+from Muon.GUI.Common.muon_pair import MuonPair
+from Muon.GUI.Common.muon_data_context import MuonDataContext
+from Muon.GUI.Common import mock_widget
+
+
+class GroupSelectorTest(unittest.TestCase):
+
+    def setUp(self):
+        self._qapp = mock_widget.mockQapp()
+        # Store an empty widget to parent all the views, and ensure they are deleted correctly
+        self.obj = QtGui.QWidget()
+
+        self.data = MuonDataContext()
+
+        self.model = PairingTableModel(data=self.data)
+        self.view = PairingTableView(parent=self.obj)
+        self.presenter = PairingTablePresenter(self.view, self.model)
+
+        self.view.warning_popup = mock.Mock()
+
+    def tearDown(self):
+        self.obj = None
+
+    def assert_model_empty(self):
+        self.assertEqual(len(self.model.pair_names), 0)
+        self.assertEqual(len(self.model.pairs), 0)
+
+    def assert_view_empty(self):
+        self.assertEqual(self.view.num_rows(), 0)
+
+    def add_three_groups_to_model(self):
+        group1 = MuonGroup(group_name="my_group_0", detector_ids=[1])
+        group2 = MuonGroup(group_name="my_group_1", detector_ids=[2])
+        group3 = MuonGroup(group_name="my_group_2", detector_ids=[3])
+        self.data.add_group(group1)
+        self.data.add_group(group2)
+        self.data.add_group(group3)
+
+    def add_two_pairs_to_table(self):
+        pair1 = MuonPair(pair_name="my_pair_0", forward_group_name="my_group_0", backward_group_name="my_group_1", alpha=1.0)
+        pair2 = MuonPair(pair_name="my_pair_1", forward_group_name="my_group_1", backward_group_name="my_group_2", alpha=1.0)
+        self.presenter.add_pair(pair1)
+        self.presenter.add_pair(pair2)
+
+    def get_group_1_selector(self, row):
```

Comment:
I think this should be called `get_pair_1_selector` to avoid confusion with the muon groups